proxy/src/coco/git_helper.rs

```diff
@@ -0,0 +1,57 @@
+//! git-remote-rad git helper related functionality.
+
+use crate::config;
+
+use crate::error;
+use std::fs;
+use std::os::unix::fs::PermissionsExt;
+use std::path::Path;
+
+/// Trait for extending the Path module.
+trait IsExecutable {
+    /// Returns true if it is an executable.
+    fn is_executable(&self) -> bool;
+}
+
+/// Implements isExecutable for Path.
+impl IsExecutable for Path {
+    fn is_executable(&self) -> bool {
+        let metadata = match self.metadata() {
+            Ok(metadata) => metadata,
+            Err(_) => return false,
+        };
+        let permissions = metadata.permissions();
+        permissions.mode() & 0o111 != 0
+    }
+}
+
+/// Checks if the git-remote-rad helper is in a stable location and has the
+/// executable flag, if not copies the executable to the right place.
+///
+/// # Errors
+///
+///   * Could not get the path to directory where helper binaries should be stored.
+///   * Could not get the current working directory.
+///   * Could not create the path to binary directory.
+///   * Could not copy helper executable to the binary directory.
+pub fn setup() -> Result<(), error::Error> {
+    log::info!("Making sure git-remote-rad helper is set up");
+
+    let helper_binary_name = "git-remote-rad";
+
+    let bin_dir = config::bin_dir()?;
+    let full_dest_path = bin_dir.join(helper_binary_name);
+
+    if full_dest_path.is_executable() {
```

Comment:
Wait, how does this confirm the file path exists?

Reply:
Or else the debug message below is confusing me

Reply:
The existence of the path was checked in the `Err` arm of this match:
https://github.com/radicle-dev/radicle-upstream/pull/769/commits/25737acf7f801d7ff182e548b8e8e02b3bc8fb06#diff-9effc3f34cdc6f654db167f9ac8ebd2eR20


Improved the readability in:
https://github.com/radicle-dev/radicle-upstream/pull/769/commits/aefa346c5b1cf35fb3073ee029d80e24cb10ea67